com.unity.shadergraph/Editor/Data/Nodes/Utility/DropdownNode.cs

```diff
@@ -0,0 +1,185 @@
+                RemoveSlot(inputSlots[i].id);
+            }
+
+            // Add output slot
+            AddSlot(new DynamicVectorMaterialSlot(OutputSlotId, "Out", "Out", SlotType.Output, Vector4.zero));
+
+            // Add input slots
+            int[] slotIds = new int[dropdown.entries.Count + 1];
+            slotIds[dropdown.entries.Count] = OutputSlotId;
+            for (int i = 0; i < dropdown.entries.Count; i++)
+            {
+                // Get slot based on entry id
+                MaterialSlot slot = inputSlots.Where(x =>
+                    x.id == dropdown.entries[i].id &&
+                    x.RawDisplayName() == dropdown.entries[i].displayName &&
+                    x.shaderOutputName == dropdown.entries[i].displayName).FirstOrDefault();
+
+                if (slot == null)
+                {
+                    slot = new DynamicVectorMaterialSlot(dropdown.entries[i].id, dropdown.entries[i].displayName, dropdown.entries[i].displayName, SlotType.Input, Vector4.zero);
+                }
+
+                AddSlot(slot);
+                slotIds[i] = dropdown.entries[i].id;
+            }
+            RemoveSlotsNameNotMatching(slotIds);
+
+            // Reconnect the edges
+            foreach (KeyValuePair<MaterialSlot, List<IEdge>> entry in edgeDict)
+            {
+                foreach (IEdge edge in entry.Value)
+                {
+                    owner.Connect(edge.outputSlot, edge.inputSlot);
+                }
+            }
+
+            ValidateNode();
+        }
+
+        public void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode generationMode)
+        {
+            var outputSlot = FindOutputSlot<MaterialSlot>(OutputSlotId);
+
+            bool isGeneratingSubgraph = owner.isSubGraph && (generationMode != GenerationMode.Preview);
+            if (generationMode == GenerationMode.Preview || !isGeneratingSubgraph)
+            {
+                sb.AppendLine(string.Format($"{outputSlot.concreteValueType.ToShaderString()} {GetVariableNameForSlot(OutputSlotId)};"));
+                var value = GetSlotValue(GetSlotIdForActiveSelection(), generationMode);
+                sb.AppendLine(string.Format($"{GetVariableNameForSlot(OutputSlotId)} = {value};"));
+            }
+            else
+            {
+                // Iterate all entries in the dropdown
+                for (int i = 0; i < dropdown.entries.Count; i++)
+                {
+                    if (i == 0)
+                    {
+                        sb.AppendLine(string.Format($"{outputSlot.concreteValueType.ToShaderString()} {GetVariableNameForSlot(OutputSlotId)};"));
+                        sb.AppendLine($"if ({m_Dropdown.value.referenceName} == {i})");
+                    }
+                    else
+                    {
+                        sb.AppendLine($"else if ({m_Dropdown.value.referenceName} == {i})");
+                    }
+
+                    {
+                        sb.AppendLine("{");
```

Comment:
nit: you could use sb.BlockScope here to clean this up a bit. In particular this becomes:
```
using(sb.BlockScope())
{
    var value = GetSlotValue(GetSlotIdForPermutation(new KeyValuePair<ShaderDropdown, int>(dropdown, i)), generationMode);
    sb.AppendLine(string.Format($"{GetVariableNameForSlot(OutputSlotId)} = {value};"));
}
```
Something I just noticed that I didn't before.